services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -0,0 +1,147 @@
+          List<Store> stores = admin.getAllStores(cluster);
+          for (Store store: stores) {
+            if (StringUtils.isEmpty(store.getTargetSwapRegion())) {
+              continue;
+            }
+
+            Map<String, Integer> coloToVersions = admin.getCurrentVersionsForMultiColos(cluster, store.getName());
+            Set<String> targetRegions = RegionUtils.parseRegionsFilterList(store.getTargetSwapRegion());
+            Set<String> remainingRegions = getRegionsForVersionSwap(coloToVersions, targetRegions);
+
+            // Do not perform version swap for davinci stores
+            int targetVersion = store.getCurrentVersion();
+            Version version = store.getVersion(targetVersion);
+            if (version.getIsDavinciHeartbeatReported()) {
+              LOGGER.info(
+                  "Skipping version swap for store: {} on version: {} as it is davinci",
+                  store.getName(),
+                  targetVersion);
+              continue;
+            }
+
+            // Check that push is completed in target regions
+            String kafkaTopicName = Version.composeKafkaTopic(store.getName(), targetVersion);
+            Admin.OfflinePushStatusInfo pushStatusInfo = admin.getOffLinePushStatus(cluster, kafkaTopicName);
+            boolean didPushCompleteInTargetRegions = true;
+            for (String targetRegion: targetRegions) {
+              String executionStatus = pushStatusInfo.getExtraInfo().get(targetRegion);
+              if (executionStatus != ExecutionStatus.COMPLETED.toString()) {
+                didPushCompleteInTargetRegions = false;
+              }
+            }
+
+            if (!didPushCompleteInTargetRegions) {
+              LOGGER.info(
+                  "Skipping version swap for store: {} on version: {} as push is not complete",
+                  store.getName(),
+                  targetVersion);
+              continue;
+            }
+
+            // Check that waitTime has elapsed in target regions
+            boolean didWaitTimeElapseInTargetRegions = true;
+            for (String targetRegion: targetRegions) {
+              Long completionTime = pushStatusInfo.getExtraInfoUpdateTimestamp().get(targetRegion);
+              Long storeWaitTime = TimeUnit.MINUTES.toMillis(store.getTargetSwapRegionWaitTime());
+              if (!(completionTime + storeWaitTime <= time.getMilliseconds())) {
+                didWaitTimeElapseInTargetRegions = false;
+              }
+            }
+
+            if (!didWaitTimeElapseInTargetRegions) {
+              LOGGER.info(
+                  "Skipping version swap for store: {} on version: {} as wait time: {} has not passed",
+                  store.getName(),
+                  targetVersion,
+                  store.getTargetSwapRegionWaitTime());
+              continue;
+            }
+
+            // Issue roll forward for remaining regions
+            for (String region: remainingRegions) {
```

Comment:
I believe `rollForwardToFutureVersion` can take the remainingRegions as region filter parameter.  ie
`rollForwardToFutureVersion(cluster, storeName, remainingRegions)` should work.